projects/plugins/crm/includes/ZeroBSCRM.InvoiceBuilder.php

```diff
@@ -621,23 +621,28 @@ function zeroBSCRM_invoicing_generateStatementHTML_v3( $contact_id = -1, $return
 
 						// ignore if status_bool (non-completed status)
 						$partial['status_bool'] = (int) $partial['status_bool'];
-						if ( isset( $partial ) && $partial['status_bool'] == 1 && isset( $partial['total'] ) && $partial['total'] > 0 ) { // phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual
-
-							// v3.0+ has + or - partials. Account for that:
-							if ( $partial['type_accounting'] === 'credit' ) {
-
-								// credit note, or refund
-								$balance = $balance + $partial['total'];
-								// add to payments
-								$payments += $partial['total'];
-
-							} else {
-
-								// assume debit
-								$balance = $balance - $partial['total'];
-
-								// add to payments
-								$payments += $partial['total'];
+						if ( isset( $partial ) && $partial['status_bool'] == 1 && isset( $partial['total'] ) ) { // phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual
+
+							switch ( $partial['type'] ) {
+								case __( 'Sale', 'zero-bs-crm' ):
+									// these count as debits against invoice.
+									$balance   = $balance - $partial['total'];
+									$payments += $partial['total'];
+
+									break;
+								case __( 'Refund', 'zero-bs-crm' ):
+								case __( 'Credit Note', 'zero-bs-crm' ):
+									// These count as credits against invoice, and should be added.
+									// If inputted as negative, then add that amount.
+									if ( str_starts_with( $partial['total'], '-' ) ) {
+										$balance = $balance + $partial['total'];
+									} else {
+										// If inputted as positive, then subtract that amount.
+										$balance = $balance - $partial['total'];
+									}
+									$payments += $partial['total'];
```

Comment:
Here we could do the same. Always add the absolute amount to the balance, ignoring the sign:

`$balance += abs( (float) $partial['total'] );`
`$payments -= abs( (float) $partial['total'] );`

Reply:
Great suggestion. I have implemented this now, however with `$balance -=`, not `+=`, as it was adding the credit / refund amount to the invoice total. Perhaps I misunderstood what we talked about though - in this case with an invoice at £100, and a credit note of - or + £5, the total is now £95.